contracts/0.8.9/StakingRouter.sol

```diff
@@ -0,0 +1,546 @@
+        uint256 _modulesCount = _modulesCache.length;
+
+        /// @dev return empty response if there are no modules or active keys
+        if (_modulesCount == 0 || _totalActiveKeys == 0) {
+            return (new address[](0), new uint16[](0), 0);
+        }
+
+        recipients = new address[](_modulesCount);
+        moduleFees = new uint16[](_modulesCount);
+        totalFee = 0;
+
+        uint256 rewardedModulesCount = 0;
+        uint256 moduleKeysShare;
+        for (uint256 i = 0; i < _modulesCount; ++i) {
+            /// @dev skip modules which have no active keys
+            if (_modulesCache[i].activeKeysCount > 0) {
+                moduleKeysShare = ((_modulesCache[i].activeKeysCount * TOTAL_BASIS_POINTS) / _totalActiveKeys);
+
+                recipients[i] = address(_modulesCache[i].stakingModuleAddress);
+                moduleFees[i] = uint16((moduleKeysShare * _modulesCache[i].moduleFee) / TOTAL_BASIS_POINTS);
+
+                totalFee += uint16((moduleKeysShare * _modulesCache[i].treasuryFee) / TOTAL_BASIS_POINTS) + moduleFees[i];
+                unchecked {
+                    rewardedModulesCount++;
+                }
+            }
+        }
+
+        /// @dev shrink arrays
+        if (rewardedModulesCount < _modulesCount) {
+            uint256 trim = _modulesCount - rewardedModulesCount;
+            assembly {
+                mstore(recipients, sub(mload(recipients), trim))
+                mstore(moduleFees, sub(mload(moduleFees), trim))
+            }
+        }
+    }
+
+    /// @notice returns new deposits allocation after the distribution of the `_keysToAllocate` keys
+    function getKeysAllocation(uint256 _keysToAllocate) public view returns (uint256 allocated, uint256[] memory allocations) {
+        (allocated, allocations, ) = _getKeysAllocation(_keysToAllocate, false);
+    }
+
+    /**
+     * @dev Invokes a deposit call to the official Deposit contract
+     * @param _maxDepositsCount max deposits count
+     * @param _stakingModuleId id of the staking module to be deposited
+     * @param _depositCalldata module calldata
+     */
+    function deposit(
+        uint256 _maxDepositsCount,
+        uint24 _stakingModuleId,
+        bytes calldata _depositCalldata
+    )
+        external
+        onlyRole(STAKING_ROUTER_DEPOSIT_ROLE)
+        onlyRegisteredStakingModule(_stakingModuleId)
+        onlyActiveStakingModule(_stakingModuleId)
+        returns (uint256)
+    {
+        /// @todo make more optimal calc of totalActiveKeysCount (eliminate double calls of module.getTotalUsedKeys() and
+        ///       module.getTotalStoppedKeys() inside getTotalActiveKeys() and _loadStakingModuleCache() methods)
+
+        uint256 maxSigningKeysCount;
+        uint256 stakingModuleIndex;
+        StakingModuleCache[] memory _modulesCache;
+        {
+            uint256[] memory newKeysAllocation;
+            _maxDepositsCount = Math.min(address(this).balance / DEPOSIT_SIZE, _maxDepositsCount);
```

Comment:
Calc alloaction based on depositable ether for module and not relay on `_maxDepositsCount `